src/libstd/panicking.rs

```diff
@@ -149,6 +149,18 @@ impl<'a> PanicInfo<'a> {
         self.payload
     }
 
+    /// Returns the payload as a slice if it is a `&'static str` or `String`.
```

Comment:
We probably can't get `&Any` casting to `&Display` to ever work, the point of `&Any` is that the information about `Display` is erased, so we can only ever recover it on a best-effort basis (e.g. checking common cases). The case where this could work is casting `&(Any + Display)` to `&Display`, but that also doesn't work today.

I personally feel like this API is kinda niche, it arguably should return `&Display` as all you really want is a human readable message out of the `Box<Any>` so we could extend that to integers, perhaps vectors, etc.